riscv/insns/ssamoswap_w.h

```diff
@@ -0,0 +1,18 @@
+require_extension(EXT_ZICFISS);
+require_extension('A');
+
+reg_t prv = STATE.prv;
+
+if (prv != PRV_M && !get_field(STATE.menvcfg->read(), MENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+if (!p->extension_enabled('S'))
+  throw trap_illegal_instruction(insn.bits());
+else if (prv == PRV_U && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else if (prv == PRV_S && STATE.v && !get_field(STATE.henvcfg->read(), HENVCFG_SSE))
+  throw trap_virtual_instruction(insn.bits());
+else if (prv == PRV_U && STATE.v && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_virtual_instruction(insn.bits());
+else
```

Comment:
This code sequence that throws the traps can be replaced by "require_envcfg(SSE)"
```
require_envcfg(SSE)
WRITE_RD(sext32(MMU.amo<uint32_t>(RS1, [&](uint32_t UNUSED lhs) { return RS2; })));
```